### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by Page-**

### CODE REVIEW

File: src/platform/errors.ts [hunk 1]
```diff
@@ -10,10 +11,19 @@ export const translateError = (err: Error | number | string): string => {
 	if (err instanceof InternalRequestError) {
 		return 'Server Error';
 	}
-	if (err instanceof Error) {
-		return err.message;
+	let message;
+	if (err instanceof HttpError) {
+		captureException(
+			new Error(),
+			'Translating a HttpError, this should go via handleHttpErrors instead',
```
Comment (dfunckt) [inline src/platform/errors.ts:18]:
```suggestion
			'Translating an HttpError, this should go via handleHttpErrors instead',
```

File: src/routes/applications.ts [hunk 2]
```diff
@@ -64,8 +64,7 @@ export const downloadImageConfig: RequestHandler = async (req, res) => {
 		res.json(config);
 	} catch (err) {
 		if (err instanceof UnauthorizedError) {
-			res.status(404).send(err.message);
-			return;
+			err = new BadRequestError(err);
```
Comment (dfunckt) [inline src/routes/applications.ts:67]: Why change the response? Shouldn't this be NotFoundError instead?

File: src/platform/errors.ts (hunk 1 above)
Comment (Page-) [reply src/platform/errors.ts:18]: It depends on your accent @dfunckt, if you say `haitch` it's `a`, if you say `aitch` it's `an` - I'd use `a`

File: src/platform/errors.ts [hunk 3]
```diff
@@ -10,10 +11,19 @@ export const translateError = (err: Error | number | string): string => {
...
+		);
+		message = err.message;
+	} else if (err instanceof Error) {
+		message = err.message;
+	} else {
+		message = `${err}`;
 	}
-	return `${err}`;
+	return escapeHtml(message);
```
Comment (thgreasi) [inline src/platform/errors.ts:26]: I hope that this doesn't make the API generated errors look weird in the dashboard.

File: src/routes/applications.ts (hunk 2 above)
Comment (Page-) [reply src/routes/applications.ts:67]: oh ffs yes

Comment (Page-) [reply src/routes/applications.ts:67]: fixed

File: src/platform/errors.ts (hunk 1 above)
Comment (dfunckt) [reply src/platform/errors.ts:18]: Okay, I think I'll trust you on this :)

File: src/platform/errors.ts (hunk 3 above)
Comment (Page-) [reply src/platform/errors.ts:26]: I would expect not, but if so we can handle those as we spot them :)

File: src/platform/errors.ts (hunk 1 above)
Comment (Page-) [reply src/platform/errors.ts:18]: Yeah it really comes down to accent, it's `an` before vowel sounds and `a` before non-vowel sounds, and the letter `H` can be said with either so the correct one depends on the reader.  You can also get vowels with non-vowel sounds like `a ukulele`

**Pull request merged**